aicsimageio/readers/lif_reader.py

```diff
@@ -0,0 +1,751 @@
+
+        Parameters
+        ----------
+        scene: int
+            The scene to retrieve the dimensions from
+
+        Returns
+        -------
+        (X, Y, Z) in µm.
+
+        """
+        # find all the Image nodes in the xml tree. They correspond to the individual scenes
+        img_sets = self.metadata.findall(".//Image")
+
+        # select the specified scene
+        img = img_sets[scene]
+        scene_pixel_size = [1.0, 1.0, 1.0]
+        dim_list = []
+        dim_list.append(img.findall(".//DimensionDescription[@DimID='1']")[0])
+        dim_list.append(img.findall(".//DimensionDescription[@DimID='2']")[0])
+        z_dim = img.findall(".//DimensionDescription[@DimID='3']")
+        if len(z_dim) > 0:
+            dim_list.append(z_dim[0])
+
+        # calculate and overwrite the pixel size for each X, Y, Z dim if present
+        for idx, dim in enumerate(dim_list):
+            # the formula for the pixel size is
+            # pixel_size = Length/(NumberOfElements - 1) from Leica & ImageJ
+            scene_pixel_size[idx] = (abs(1000000.0*float(dim.attrib['Length']))/(
+                                        float(dim.attrib['NumberOfElements'])-1.0)
+                                     )
+        return tuple(scene_pixel_size)
+
+    #  bitmap reader functions
+
+    @staticmethod
+    def _get_item_as_bitmap(im_path: Path, offsets: List[type(np.ndarray)], r_length: np.ndarray,
+                            read_dims: Optional[Dict[str, int]] = None) -> Tuple[List[type(np.ndarray)],
+                                                                                 List[Tuple[str, int]]]:
+        """
+        Gets specified bitmap data from the lif file (private).
+
+        Parameters
+        ----------
+        im_path: Path
+            Path to the LIF file to read.
+        offsets: List[numpy.ndarray]
+            A List of numpy ndarrays offsets, see _compute_offsets for more details.
+        r_length: numpy.ndarray
+            A 1D numpy array of read lengths, the index is the scene index
+        read_dims: Optional[Dict[str, int]]
+            The dimensions to read from the file as a dictionary of string to integer.
+            Default: None (Read all data from the image)
+
+        Returns
+        -------
+        numpy.ndarray
+            a stack of images as a numpy.ndarray
+        List[Tuple[str, int]]
+            The shape of the data being returned
+        """
+        if read_dims is None:
+            read_dims = {}
+
+        lif = LifFile(im_path)
+
+        # data has already been checked for consistency. The dims are either consistent or S is specified
+        selected_ranges = LifReader.read_dims_to_ranges(lif, read_dims)
+        s_index = read_dims[Dimensions.Scene] if Dimensions.Scene in read_dims.keys() else 0
+        lif_img = lif.get_image(img_n=s_index)
+        x_size = lif_img.dims[0]
+        y_size = lif_img.dims[1]
+        lir = LifReader(im_path)
```

Comment:
I am a bit confused... are you creating a `LifReader` object _inside_ a `LifReader` static method? I worry there may be some odd consequences of multiple indexing / memory. I would just call `LifReader.get_pixel_type`, no need to instantiate an entire `LifReader` object.

Reply:
good point, I implemented your suggestion.